dissect/target/helpers/modifier.py

```diff
@@ -0,0 +1,98 @@
+import warnings
+from functools import partial
+from typing import BinaryIO, Callable, Iterator
+
+from flow.record import GroupedRecord, Record, RecordDescriptor, fieldtypes
+
+from dissect.target import Target
+from dissect.target.exceptions import FilesystemError
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.hashutil import common
+from dissect.target.helpers.utils import StrEnum
+
+RECORD_NAME = "filesystem/file/digest"
+NAME_SUFFIXES = ["_resolved", "_digest"]
+RECORD_TYPES = ["path", "digest"]
+
+
+def _resolve_path_types(target: Target, record: Record) -> Iterator[tuple[TargetPath, str]]:
+    for field_name, field_type in record._field_types.items():
+        if not issubclass(field_type, fieldtypes.path):
+            continue
+
+        path = getattr(record, field_name, None)
+        if path is None:
+            continue
+
+        yield target.resolve(str(path)), field_name
```

Comment:
```suggestion
        yield field_name, target.resolve(str(path))
```

This order feels more logical (it is more in line with key, value pairs).